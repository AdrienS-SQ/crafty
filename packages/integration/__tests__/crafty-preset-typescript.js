/* global describe, it, expect */

const fs = require("fs");
const path = require("path");

const rimraf = require("rimraf");

const testUtils = require("../utils");

it("Lints TypeScript using the command", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-typescript/lints")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["tsLint", "js/**/*.ts"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});
