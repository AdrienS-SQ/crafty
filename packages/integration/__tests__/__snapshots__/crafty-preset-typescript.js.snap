// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lints TypeScript using the command 1`] = `
Object {
  "status": 1,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/Component.ts
  2:1   error  Prefer default export                         @swissquote/swissquote/import/prefer-default-export
  3:1   error  Expected indentation of 2 spaces but found 3  indent
  3:18  error  Multiple spaces found before 'b'              no-multi-spaces

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/script.ts
   6:1   error    Expected indentation of 2 spaces but found 4                                   indent
   7:5   warning  Unexpected console statement                                                   no-console
   8:1   error    Expected indentation of 2 spaces but found 1                                   indent
  11:5   warning  Unexpected console statement                                                   no-console
  14:5   error    Opening curly brace does not appear on the same line as controlling statement  brace-style
  17:7   warning  Unexpected console statement                                                   no-console
  18:7   error    There should be no space before ','                                            comma-spacing
  18:13  error    Missing semicolon                                                              semi

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/SomeLibrary.ts
   3:1  error  Expected indentation of 2 spaces but found 4                                   indent
   4:1  error  Expected indentation of 4 spaces but found 8                                   indent
   5:1  error  Expected indentation of 2 spaces but found 4                                   indent
   7:1  error  Expected indentation of 2 spaces but found 4                                   indent
   8:1  error  Expected indentation of 4 spaces but found 2                                   indent
   9:1  error  Expected indentation of 2 spaces but found 4                                   indent
  11:1  error  Expected indentation of 2 spaces but found 4                                   indent
  12:1  error  Expected indentation of 2 spaces but found 4                                   indent
  12:5  error  Opening curly brace does not appear on the same line as controlling statement  brace-style
  13:1  error  Expected indentation of 4 spaces but found 8                                   indent
  14:1  error  Expected indentation of 2 spaces but found 6                                   indent

âœ– 22 problems (19 errors, 3 warnings)
  18 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;
